modules/renter/siafile/metadata.go

```diff
@@ -0,0 +1,288 @@
+package siafile
+
+import (
+	"math"
+	"os"
+	"time"
+
+	"github.com/NebulousLabs/Sia/build"
+	"github.com/NebulousLabs/Sia/crypto"
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+type (
+	// Metadata is the metadata of a SiaFile and is JSON encoded.
+	Metadata struct {
+		version      [16]byte          // version of the sia file format used
+		fileSize     int64             // total size of the file
+		masterKey    crypto.TwofishKey // masterkey used to encrypt pieces
+		pieceSize    uint64            // size of a single piece of the file
+		trackingPath string            // file to the local copy of the file used for repairing
+		siaPath      string
+
+		// The following fields are the usual unix timestamps of files.
+		modTime    time.Time // time of last content modification
+		changeTime time.Time // time of last metadata modification
+		accessTime time.Time // time of last access
```

Comment:
are we intending to have the accessTime and the downloadTime be the same?

My guess is not, does that mean that we should add a downloadTime as well?

Reply:
I think just having the `accessTime` is fine since that is also the way Unix does it. `modTime` is bascially a subset of `accessTime` since it only includes uploads. 
If you just want to know the last time you manually downloaded a file, you can still take a look at the download history.

Reply:
hmm.

Doesn't a download count as an access though? You did technically access the file if you performed a download.

And, are we treating these only as applies to the logical data? Because that means that repairing would not update any of these files at all (which I think is correct)

Reply:
Yeah, a download would be an access. Once we support updating the file we would also reset `modTime` when we do so.
I'm not sure about repairing since it also downloads the file. Would make sense for it not to change the timestamp though.